utils.py

```diff
@@ -77,3 +79,159 @@ def decorated_function(*args, **kwargs):
         return decorated_function
 
     return decorator
+
+
+'''
+    Requirements for auth:
+        SwagLyrics PEM should be saved as "swag_pem" in environment variables
+        APP_ID is the id of the github app
+        repo_url is the api url of the repo this app is installed on
+'''
+
+APP_ID = os.getenv('swag_appid')
+
+# WARN: Security vulnerabilities - auth methods should eventually be moved
+# to its own class
+expiry_time, jwtoken = None, None
+
+repo_url = "https://api.github.com/repos/SwagLyrics/SwagLyrics-For-Spotify/"
+
+
+def produce_jwt() -> bytes:
+    """
+        Creates a JWT(Json Web Token) based on the downloaded key saved as
+        "swag_pem" in user environment variables
+
+        :return: The produced JWT if it can be produced, else None
+    """
+
+    # Can be read as string, but bytes better corresponds to what should go in
+    # api calls
+    with open(os.environ["swag_pem"], "rb") as f:
+        private_pem = f.read()
+    global jwtoken, expiry_time
+    now = int(time())
+
+    expiry_time = now + 10 * 60  # 10 minutes
+    payload = {
+        "iat": now,
+        "exp": expiry_time,
+        "iss": APP_ID
+    }
+    try:
+        jwtoken = jwt.encode(payload, private_pem, "RS256")
+    except ValueError as v_err:
+        print("Value Error: {}".format(v_err))
+        return
+
+    print(f"This token expires on {ctime(expiry_time)}")
+    return jwtoken
+
+
+def authenticate_github_app(jwtoken: bytes):
+    """
+        Authenticates the github app with the JWT
+
+        :param: JWT
+        :return: Success if authentication succeeded, failed otherwise
+    """
+    github_app_url = "https://api.github.com/app"
+    # print(type(jwt))
+    headers = genheaders(jwtoken=get_jwt())
+
+    # print(f"bearer: {bearer}")
+    res = requests.get(
+        github_app_url,
+        headers=headers)
+
+    if res.status_code == 200:
+        print("Success - token authorized")
+        return 0
+    else:
+        print(f"Github App Auth failed with status code: {res.status_code}")
+        return 1
+
+
+def get_jwt() -> bytes:
+    """
+        A method to ensure we only authorize when needed
+
+        :return: An authorized JWT
+    """
+
+    global jwtoken, expiry_time
+
+    if not jwtoken or time() > expiry_time:
```

Comment:
In generally I don't like cutting it too close so maybe we could take a buffer of 30 seconds?